domains/grid/utils/gridParser.ts

```diff
@@ -135,14 +135,16 @@ const parseSoundCloudWidgetInput = (
 const parseInstagramWidgetInput = (
   input: string
 ): LayoutItemExtended | never => {
-  // TODO: Add support for instagram timeline embeds when we have the widget type
-  const INSTAGRAM_PERMALINK_REGEX = /data-instgrm-permalink="([^"]+)"/
-  const instagramMatch = input.match(INSTAGRAM_PERMALINK_REGEX)
-  if (instagramMatch) {
+  const INSTAGRAM_URL_REGEX =
+    /https:\/\/www\.instagram\.com\/(p|reel|profile|tv)\/[\w-]+\/(?:\?[^"]*)?/
+
+  const match = input.match(INSTAGRAM_URL_REGEX)
```

Comment:
Can you use spread to name match parts like I did in other parsers? I'm not a fan of using indexes as they hide the purpose of the value.